pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -673,19 +676,21 @@ public void createConsumingSegment(String realtimeTableName, List<Integer> nonCo
           LOGGER.info("Creating CONSUMING segment for {} partition {} with seq {}", realtimeTableName, curPartition, nextSeqNum);
           long startOffset = getPartitionOffset(kafkaStreamMetadata, "smallest", curPartition);
```

Comment:
Should this be smallest or the offset configured in the table? For example, if you configure a table with largest offset and you delete the only segment, then you'd restart consumption with the smallest offset, which might not be what the user expects.